batch-models/src/main/scala/org/ekstep/analytics/dashboard/CompetencyMetricsModel.scala

```diff
@@ -321,9 +322,131 @@ object CompetencyMetricsModel extends IBatchModelTemplate[String, DummyInput, Du
     // mdo-wise courses completed at-least once
     val liveCourseCompletedAtLeastOnceByMDODF = liveCourseCompletedDF.groupBy("userOrgID").agg(countDistinct("courseID").alias("count"))
     redisDispatchDataFrame[Long]("dashboard_courses_completed_at_least_once_by_user_org", liveCourseCompletedAtLeastOnceByMDODF, "userOrgID", "count")
-
     // new redis updates - end
 
+    //get the current date, the previous calendar date and a day before that for learner stats - start
+    val currentDateDF = spark.range(1).select(current_date().alias("currentDate"))
+    // Format the current date as a string in the desired format
+    val dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSSZ")
+    val currentDateString = currentDateDF.select(date_format($"currentDate", "yyyy-MM-dd HH:mm:ss.SSSSSSZ").alias("formattedCurrentDate")).first().getString(0)
+    // Calculate the start and end of the current day with timezone offset "0000"
+    val endOfCurrentDay = currentDateDF.select(to_utc_timestamp(concat($"currentDate", lit(" 23:59:59.896")), "0000").alias("endOfCurrentDay")).first().getTimestamp(0)
+    // Calculate the start and end of the day 7 days ago (a week) with timezone offset "0000"
+    val startOf7thDay = currentDateDF.select(to_utc_timestamp(concat($"currentDate" - expr("INTERVAL 7 DAY"), lit(" 00:00:00")), "0000").alias("startOf7thDay")).first().getTimestamp(0)
+
+    // Format start and end of the current and previous day as strings
+    val endOfCurrentDayString = dateFormat.format(endOfCurrentDay)
+    val startOf7thDayString = dateFormat.format(startOf7thDay)
+    print(endOfCurrentDayString)
+    print(startOf7thDayString)
+    //get the current date, the previous calendar date and a day before that for learner stats - end
+
+    // learner home page data logic - start
+    print("started calculating")
+    val lastRunDate: String = redisGetKeyValue("lhp_lastRunDate")
+    print("The last run date is "+ lastRunDate)
+    print("current Date is"+ currentDateString)
+    print(lastRunDate.equals(currentDateString))
+    if(!(lastRunDate.equals(currentDateString)))
+      {
+        val totalLearningHoursTillTodayByOrg = allCourseProgramCompletionWithDetailsDF.groupBy("userOrgID").agg(sum(expr("(completionPercentage / 100) * courseDuration")).alias("totalLearningHours"))
```

Comment:
handle the case where courseDuration=0, set the learning hours to 0 as well, this should not happen in prod data, but our code should be robust enough to handle this case